packages/cloud-cognitive/src/components/SidePanel/SidePanel.js

```diff
@@ -706,86 +693,98 @@ export let SidePanel = React.forwardRef(
+              ref={contentRef}
+              role="complementary"
+              aria-label={title}
+              onAnimationComplete={onAnimationEnd}
+              onAnimationStart={(event) => onAnimationStart(event)}
+              initial={{
+                x: placement === 'right' ? '100%' : -320,
               }}
-            />
-          )}
-        </>
-      )
+              animate={{
+                x: 0,
+                transition: {
+                  duration: DURATIONS.moderate02,
+                  ease: EASINGS.productive.standard,
+                },
+              }}
+              exit={{
+                x: placement === 'right' ? '100%' : -320,
+                transition: {
+                  duration: DURATIONS.moderate01,
+                  ease: EASINGS.productive.exit,
+                },
+              }}
+            >
+              <span
+                ref={startTrapRef}
+                tabIndex="0"
+                role="link"
+                className={`${blockClass}__visually-hidden`}
+              >
+                Focus sentinel
+              </span>
+              {!animateTitle && renderHeader()}
+              <div
+                ref={sidePanelInnerRef}
+                className={cx(`${blockClass}__inner-content`, {
+                  [`${blockClass}__static-inner-content`]: !animateTitle,
+                  [`${blockClass}__static-inner-content-no-actions`]:
+                    !animateTitle && !actions?.length,
+                  [`${blockClass}__inner-content-with-actions`]:
+                    actions && actions.length,
+                })}
+              >
+                {animateTitle && renderHeader()}
+                <div className={`${blockClass}__body-content`}>{children}</div>
+                <ActionSet
+                  actions={actions}
+                  className={primaryActionContainerClassNames}
+                  size={size}
+                />
+              </div>
+              <span
+                ref={endTrapRef}
+                tabIndex="0"
+                role="link"
+                className={`${blockClass}__visually-hidden`}
+              >
+                Focus sentinel
+              </span>
+            </motion.div>
+            <AnimatePresence>
+              {includeOverlay && (
+                <motion.div
+                  initial={{
+                    opacity: 0,
+                  }}
+                  animate={{
+                    opacity: 1,
+                    transition: {
+                      duration: DURATIONS.moderate02,
+                      ease: EASINGS.productive.standard,
+                    },
+                  }}
+                  exit={{
+                    opacity: 0,
+                  }}
+                  ref={sidePanelOverlayRef}
+                  className={`${blockClass}__overlay`}
+                />
+              )}
+            </AnimatePresence>
+          </>
+        )}
+      </AnimatePresence>
```

Comment:
I was thinking the animation adds alot of code to the overall component. We should utilize variants and have them in a motion folder in the root of `Sidepanel` that way our SidePanel code is leaner and less busy with other things

Reply:
Added `SidePanel/motion/variants.js` [here](https://github.com/carbon-design-system/ibm-cloud-cognitive/pull/2312/commits/71e71372aca5145173362a6011cb36d0c8226067#diff-0c22a7c94a79e71402a58d13ac72ddd161d3d4325b589d9982317ed4485ff38d)